### PR TITLE
Fixes #4522

### LINKED ISSUE #4522: ipfsapi library is deprecated
**Describe the bug**
web_1      | /code/app/dashboard/utils.py:27: FutureWarning: The `ipfsapi` library is deprecated and will stop receiving updates on the 31.12.2019! If you are on Python 3.5+ please enable and fix all Python deprecation warnings (CPython flag `-Wd`) and switch to the new `ipfshttpclient` library name. Python 2.7 and 3.4 will not be supported by the new library, so please upgrade.
web_1      |   import ipfsapi

**To Reproduce**
start docker

**Expected behavior**
no outdated libraries

### PR DESCRIPTION
**Pull request by acolytec3**
##### Description

Replace deprecated ifpsapi library with ipfshttpclient in dashboard, kudos, and healthcheck..  Add tests for dashboard.utils that interact with IPFS.

##### Refers/Fixes

Fixes #4522'

##### Testing

Added tests for get_ipfs and ipfs_cat_ipfsapi in dashboard.utils to verify connectivity and retrieval of objects by IPFS hash string are working.

### CODE REVIEW

**Comment by acolytec3** [comment]
Hold on this.  I thought everything was working and now I'm seeing some errors in my local build.

**Comment by acolytec3** [comment]
Appears to be working now that I've corrected the connect parameters.  I'm not seeing the connection errors that surfaced. previously so should be ready for review.

**Comment by acolytec3** [comment]
@owocki @kuhnchris Any chance I could get this reviewed?  Would love to get it closed so I can move on a look at applying for some other bounties.

Thanks!

**Comment by kuhnchris** [comment]
@danlipert do you have time to review this? Else I'll take a look

**Comment by danlipert** [comment]
@kuhnchris @acolytec3 The existing tests are pretty lackluster and don't really test IPFS. I'd like some a better writeup of how this change was tested and what the differences are between this library and the newer one. Is it simply a rename? Since there are compatibility issues with Python 3.4 I'm guessing its not.

**Comment by acolytec3** [comment]
[Here's](https://github.com/ipfs/py-ipfs-api#important-changes-from-ipfsapi-04x) the official write-up on the changes from the old library.

Based on reading the docs, the only change that's pertinent for how gitcoin uses IPFS is the connect method and I've updated that to match the new structure in both of the places in the code where it's called.

The other functions from the library that are used (cat, get_json, add_json) appear to be unchanged.  

I'll look at writing some tests for those specific utils methods where connect is used.

**Comment by acolytec3** [comment]
@danlipert I added tests for both of the dashboard utility methods that call ipfs.  Let me know if this is what you're looking for.

File: app/dashboard/utils.py [hunk 1]
```diff
@@ -187,7 +187,8 @@ def get_ipfs(host=None, port=settings.IPFS_API_PORT):
 
     Args:
         host (str): The IPFS host to connect to.
-            Defaults to environment variable: IPFS_HOST.
+            Defaults to environment variable: IPFS_HOST.  The host name should be of the form 'ipfs.infura.io' and not
```
Comment (danlipert) [inline app/dashboard/utils.py]: nice add :)

Comment (acolytec3) [reply app/dashboard/utils.py]: This is from the Django shell.  I think I hit all the functions that the app uses (connecting to the infura gateway, retrieving files from a hash, adding/retrieiving json).   Is this what you're thinking?

ipfs interactions

Comment (danlipert) [reply app/dashboard/utils.py]: I meant more like running the app locally, and then sending around some kudos, creating some test bounties, etc.

Comment (acolytec3) [reply app/dashboard/utils.py]: Oh, sorry, misunderstood.  I hadn't previously but just did.  Sent a couple of kudos and posted 2 new test bounties and seemed to work fine.  

Logs from sending kudos - I don't see any errors (though not 100% sure what I'd be looking for)
> web_1      | 172.19.0.1 - - [22/Jun/2019 10:47:42] "POST /kudos/send/3/ HTTP/1.1" 200 -
web_1      | WARNING 2019-06-22 10:47:45,791 notifications 534 140604666288872 Notifications are disabled.  Skipping email.
web_1      | 172.19.0.1 - - [22/Jun/2019 10:47:45] "POST /kudos/send/4/ HTTP/1.1" 200 -
web_1      | 172.19.0.1 - - [22/Jun/2019 10:47:53] "GET /kudos/marketplace/ HTTP/1.1" 200 -

Logs from posting bounty - I do see that one error but but I believe that's the bounty posting code complaining about me not populating some value in the bounty form - doesn't look like an IPFS issue though
> web_1      | 172.19.0.1 - - [22/Jun/2019 10:43:54] "GET /funding/new HTTP/1.1" 200 -
web_1      | 172.19.0.1 - - [22/Jun/2019 10:43:54] "GET /sync/get_amount?amount=0.001&denomination= HTTP/1.1" 200 -
web_1      | 172.19.0.1 - - [22/Jun/2019 10:43:58] "GET /sync/get_issue_details?url=https%3A%2F%2Fgithub.com%2Facolytec3%2Fnanobitcoin-go%2Fissues%2F2&token=4d36bce0f325b13f672ce9ebabf3c07babe80b6f HTTP/1.1" 200 -
web_1      | 172.19.0.1 - - [22/Jun/2019 10:43:59] "GET /sync/get_issue_details?url=https%3A%2F%2Fgithub.com%2Facolytec3%2Fnanobitcoin-go%2Fissues%2F2&token=4d36bce0f325b13f672ce9ebabf3c07babe80b6f HTTP/1.1" 200 -
web_1      | 172.19.0.1 - - [22/Jun/2019 10:43:59] "GET /api/v0.1/get_suggested_contributors?keywords=&invite= HTTP/1.1" 200 -
web_1      | 172.19.0.1 - - [22/Jun/2019 10:43:59] "GET /api/v0.1/get_suggested_contributors?keywords=&invite= HTTP/1.1" 200 -
**web_1      | ERROR 2019-06-22 10:44:30,310 views 534 140604666288872 expected string or bytes-like object**
web_1      | 172.19.0.1 - - [22/Jun/2019 10:44:30] "GET /funding/details/?url=https://github.com/acolytec3/nanobitcoin-go/issues/2 HTTP/1.1" 200 -
web_1      | 172.19.0.1 - - [22/Jun/2019 10:44:30] "GET /jsi18n/ HTTP/1.1" 200 -
web_1      | 172.19.0.1 - - [22/Jun/2019 10:44:30] "GET /dynamic/js/tokens_dynamic.js HTTP/1.1" 200 -
web_1      | 172.19.0.1 - - [22/Jun/2019 10:44:30] "GET /media/avatars/be43ca7feb8423165e8300b1e6f99877/acolytec3.png HTTP/1.1" 304 -
web_1      | 172.19.0.1 - - [22/Jun/2019 10:44:31] "GET /inbox/notifications/?page=1 HTTP/1.1" 200 -
**web_1      | ERROR 2019-06-22 10:44:31,572 views 534 140604666288872 expected string or bytes-like object**
web_1      | 172.19.0.1 - - [22/Jun/2019 10:44:31] "GET /funding/details/?url=https://github.com/acolytec3/nanobitcoin-go/issues/2 HTTP/1.1" 200 -
web_1      | * confirming tx has mined
web_1      | * getting bounty id
web_1      | ** get_bounty_id_from_web3; looking at 2210
web_1      | * getting bounty
web_1      | * processing bounty

Skipping all the json from the bounties...
>web_1      | - processing changes, None => 6
web_1      | - new_bounty event; diff => None
web_1      | ============ posting ==============
web_1      | ============ done posting ==============
web_1      | changes processed: 
web_1      | {   'did_bsr': False,
web_1      |     'did_post_to_email': False,
web_1      |     'did_post_to_github': False,
web_1      |     'did_post_to_slack': False,
web_1      |     'did_post_to_twitter': False,
web_1      |     'did_post_to_user_discord': False,
web_1      |     'did_post_to_user_slack': False}
web_1      | 172.19.0.1 - - [22/Jun/2019 10:44:51] "POST /sync/web3/ HTTP/1.1" 200 -

Does anything jump out at you as an IPFS related issue?

Comment (kuhnchris) [reply app/dashboard/utils.py]: êasiest would be to check if the bounty/kudos/grant has a properly filled ipfs link - if that's the case and it's openable via ipfs.infura.io then it's fine.

**Comment by acolytec3** [comment]
Ah, got it. Let me rebuild that branch when I have a chance and I'll check
and confirm.

File: app/dashboard/utils.py (hunk 1 above)
Comment (acolytec3) [reply app/dashboard/utils.py]: @danlipert @kuhnchris   Finally found what I was looking for. [This](https://ipfs.infura.io/ipfs/QmaL1ve4XWwt4tU56DKPajHymDACBDHa4W6EVcB7duR11H) is the IPFS hash of a test bounty I created while running this branch locally.  Can we call this one resolved?

Comment (danlipert) [reply app/dashboard/utils.py]: Awesome @acolytec3 looks great!

Comment (kuhnchris) [reply app/dashboard/utils.py]: LGTM

Comment (acolytec3) [reply app/dashboard/utils.py]: Cool, thanks!

**Comment by acolytec3** [comment]
@danlipert @octavioamu @thelostone-mc @owocki Hi all, what's the status on this one? Are there more tests or validations you're looking for to close this out?

**Comment by owocki** [comment]
quick question.. are PRs that are approved included in releases automatically, or is there some other inclusion metric?

**Comment by danlipert** [comment]
> quick question.. are PRs that are approved included in releases automatically, or is there some other inclusion metric?

Once they are approved by each engineer, then the final approving engineer usually merges it. Sometimes we save merging until before the deploy in case there are conflicts, especially with migrations.

@thelostone-mc @octavioamu this look good to y'all?

**Comment by acolytec3** [comment]
What about this one? Is it good to go? Would love to get paid if it's done. 😬

**Comment by kuhnchris** [comment]
LGTM too - works locally at least

**Pull request merged**